ovmf/tdx_data.go

```diff
@@ -119,17 +119,30 @@ func validateTDXMetadataSections(firmwareLen uint32, rawMetadata *abi.TDXMetadat
 			}
 			foundTDHOB = true
 		case abi.TDXMetadataSectionTypeTempMem: // do nothing
+		case abi.TDXMetadataSectionTypePayload: // do nothing
+		case abi.TDXMetadataSectionTypeTDINFO: // do nothing
+		case abi.TDXMetadataSectionTypePayloadParam:
+			if section.DataSize != 0 {
+				return fmt.Errorf("PayloadParam section type had data size 0x%x. Want 0.",
+					section.DataSize)
+			}
+		case abi.TDXMetadataSectionTypePermMem:
+			if section.DataSize != 0 {
+				return fmt.Errorf("PermMem section type had data size 0x%x. Want 0.",
+					section.DataSize)
+			}
+			foundPermMem = true
 		default:
 			return fmt.Errorf("unsupported metadata section type: %v", section.SectionType)
 		}
 	}
-	if !foundTDHOB {
+	if !foundTDHOB && (!foundPermMem || !allowTDShim) {
```

Comment:
Can you explain this line? I don't get what it's trying to check

Reply:
Done.